src/test/java/com/hivemq/extensions/callbacks/DnsClusterDiscoveryTest.java

```diff
@@ -107,4 +109,28 @@ public void testDnsClusterDiscoverySingleNode() throws Exception {
         assertEquals("212.72.72.12", output.toString());
     }
 
+    @Test
+    public void testBackOff() throws Exception {
+        final String discoveryAddress = "www.hivemq.com";
+        final int discoveryTimeout = 30;
+
+        when(configuration.discoveryAddress()).thenReturn(discoveryAddress);
+        when(configuration.resolutionTimeout()).thenReturn(discoveryTimeout);
+        when(configuration.initialDiscoveryInterval()).thenReturn(1);
+        when(configuration.maxDiscoveryInterval()).thenReturn(60);
+
+        dnsClusterDiscovery.init(input, output);
+        assertEquals(2, nextReload.get());
+        dnsClusterDiscovery.reload(input, output);
+        assertEquals(4, nextReload.get());
+        dnsClusterDiscovery.reload(input, output);
+        assertEquals(8, nextReload.get());
+        dnsClusterDiscovery.reload(input, output);
+        assertEquals(16, nextReload.get());
+        dnsClusterDiscovery.reload(input, output);
+        assertEquals(32, nextReload.get());
+        dnsClusterDiscovery.reload(input, output);
+        assertEquals(60, nextReload.get());
```

Comment:
Add this line again to check we don't go over the limit:
dnsClusterDiscovery.reload(input, output);
assertEquals(60, nextReload.get());